airbyte-integrations/connectors/destination-s3/src/main/resources/spec.json

```
@@ -348,42 +340,59 @@
                 "default": "UNCOMPRESSED"
               },
               "block_size_mb": {
-                "title": "Block Size (Row Group Size) (MB)",
+                "title": "Block Size (Row Group Size) (MB) (Optional)",
                 "description": "This is the size of a row group being buffered in memory. It limits the memory usage when writing. Larger values will improve the IO when reading, but consume more memory when writing. Default: 128 MB.",
                 "type": "integer",
                 "default": 128,
                 "examples": [128]
               },
               "max_padding_size_mb": {
-                "title": "Max Padding Size (MB)",
+                "title": "Max Padding Size (MB) (Optional)",
                 "description": "Maximum size allowed as padding to align row groups. This is also the minimum size of a row group. Default: 8 MB.",
                 "type": "integer",
                 "default": 8,
                 "examples": [8]
               },
               "page_size_kb": {
-                "title": "Page Size (KB)",
+                "title": "Page Size (KB) (Optional)",
                 "description": "The page size is for compression. A block is composed of pages. A page is the smallest unit that must be read fully to access a single record. If this value is too small, the compression will deteriorate. Default: 1024 KB.",
                 "type": "integer",
                 "default": 1024,
                 "examples": [1024]
               },
               "dictionary_page_size_kb": {
-                "title": "Dictionary Page Size (KB)",
+                "title": "Dictionary Page Size (KB) (Optional)",
                 "description": "There is one dictionary page per column per row group when dictionary encoding is used. The dictionary page size works like the page size but for dictionary. Default: 1024 KB.",
                 "type": "integer",
                 "default": 1024,
                 "examples": [1024]
               },
               "dictionary_encoding": {
-                "title": "Dictionary Encoding",
+                "title": "Dictionary Encoding (Optional)",
                 "description": "Default: true.",
                 "type": "boolean",
                 "default": true
               }
             }
           }
         ],
+        "order": 5
+      },
+      "s3_endpoint": {
+        "title": "Endpoint (Optional)",
+        "type": "string",
+        "default": "",
+        "description": "This is your S3 endpoint url.(if you are working with AWS S3, just leave empty).",
+        "examples": ["http://localhost:9000"],
+        "order": 6
+      },
+      "s3_path_format": {
+        "title": "s3_path_format (Optional)",
```

```suggestion
        "title": "S3 Path Format (Optional)",
```

Please make sure all these fields are updated to reflect the name and not the code.